match2/wikis/rocketleague/legacy_match_list.lua

```diff
@@ -0,0 +1,238 @@
+	args = LegacyMatchList.copyDetailsToArgs(args, details)
+
+	return LegacyMatchList.toEncodedJson(args)
+end
+
+--THIS FUNCTION IS NOT READY FOR USAGE
+function LegacyMatchList.convertSwissMatchMaps(frame)
+	local args = getArgs(frame)
+	local details = json.parseIfString(args.details or '{}')
+
+	--process opponents
+	for index = 1, 2 do
+		local player = args['player' .. index] or args['team' .. index] or 'TBD'
+		local playerLink = args['player' .. index .. 'link'] or player
+
+		local score
+		if args.walkover then
+			if tonumber(args.walkover) == index then
+				score = 'W'
+			elseif Table.includes(ALLOWED_STATUSES, args['games' .. index]) then
+				score = args['games' .. index]
+			else
+				score = 'L'
+			end
+		else
+			score = args['games' .. index] or '0'
+		end
+		if player ~= 'TBD' then
+			args['opponent' .. index] = MatchSubobjects.luaGetOpponent(frame, {
+					type = 'solo',
+					match2players = '[' .. json.stringify({
+						name = playerLink,
+						displayname = player,
+						flag = args['p' .. index .. 'flag'],
+					}) .. ']',
+					template = args['p' .. index .. 'team'],
+					score = score,
+				})
+		end
+		args['opponent' .. index .. 'literal'] = args['team' .. index .. 'literal']
+
+		--atm we ignore the old teamXstanding parameters, because
+		--they are not supported in the new system
+
+		--empty all the stuff we set into this opponent
+		args['player' .. index] = nil
+		args['team' .. index] = nil
+		args['p' .. index .. 'flag'] = nil
+		args['p' .. index .. 'team'] = nil
+		args['team' .. index .. 'literal'] = nil
+		args['games' .. index] = nil
+		args.walkover = nil
+	end
+
+	--process maps
+	args, details = LegacyMatchList.processMaps(args, details)
+
+	--process other stuff from details
+	args = LegacyMatchList.copyDetailsToArgs(args, details)
+
+	return LegacyMatchList.toEncodedJson(args)
+end
+
+--functions shared between convertMatchMaps and convertSwissMatchMaps
+function LegacyMatchList.copyDetailsToArgs(args, details)
+	for key, value in pairs(details) do
+		if Logic.isEmpty(args[key]) then
+			args[key] = value
+		end
+	end
+	args.details = nil
+	return args
+end
+
+function LegacyMatchList.processMaps(args, details)
+	for index = 1, 15 do
+		if details['map' .. index] then
+			args['map' .. index] = MatchSubobjects.luaGetMap(nil, {
+				map = details['map' .. index],
+				winner = details['map' .. index .. 'win'],
+				score1 = details['map' .. index .. 't1score'],
+				score2 = details['map' .. index .. 't2score'],
+				ot = details['ot' .. index],
+				otlength = details['otlength' .. index],
+				vod = details['vodgame' .. index],
+				comment = details['map' .. index .. 'comment'],
+				})
+			--atm we ignore the old mapXtYgoals parameters, because
+			--1) according to Lukasz they are not used nor disaplayed anymore anyways
+			--2) they are pretty hard to convert, due to the new system wanting goal times
+			--tied to the participants and that info isn't available for the old stuff
```

Comment:
Does this mean the old data is lost?

Reply:
kind of
i was planning on adding a tracking category to the old template if those params are used
that way we should be able to see how much data we would lose

Reply:
added storage for it
Display is another cup of tea ... (needs adjusting of MatchSummary module)